### PR TITLE
Support exclude_alerts parameter in Prometheus list rules endpoint

### PR DESCRIPTION
**Pull request by alexander-akhmetov**
#### What this PR does

Adds `exclude_alerts` GET parameter to the Prometheus rules API endpoint, same as in Prometheus ([documentation](https://prometheus.io/docs/prometheus/latest/querying/api/#rules)).

It also passes the parameter further to other replicas in the gRPC call.

<details>
  <summary><b>Example</b></summary>
  
  #### Without `exclude_alerts`
```
curl -s 'http://127.0.0.1:8001/prometheus/api/v1/rules?rule_group=group_2&rule_name=alert_0' | jq
{
  "status": "success",
  "data": {
    "groups": [
      {
        "name": "group_2",
        "file": "mimirtool",
        "rules": [
          {
            "state": "firing",
            "name": "alert_0",
            "query": "vector(1) > 0",
            "duration": 60,
            "keepFiringFor": 0,
            "labels": {
              "severity": "no"
            },
            "annotations": {
              "summary": "Always firing"
            },
            "alerts": [
              {
                "labels": {
                  "alertname": "alert_0",
                  "severity": "no"
                },
                "annotations": {
                  "summary": "Always firing"
                },
                "state": "firing",
                "activeAt": "2024-09-17T12:14:31.71939972Z",
                "value": "1e+00"
              }
            ],
            "health": "ok",
            "lastError": "",
            "type": "alerting",
            "lastEvaluation": "2024-09-17T12:17:31.722263334Z",
            "evaluationTime": 0.005084959
          }
        ],
        "interval": 30,
        "lastEvaluation": "2024-09-17T12:17:31.722165334Z",
        "evaluationTime": 0.017278292,
        "sourceTenants": null
      }
    ]
  },
  "errorType": "",
  "error": ""
}
```

  #### `exclude_alerts=true`
  ```
curl -s 'http://127.0.0.1:8001/prometheus/api/v1/rules?rule_group=group_2&rule_name=alert_0&exclude_alerts=true' | jq
{
  "status": "success",
  "data": {
    "groups": [
      {
        "name": "group_2",
        "file": "mimirtool",
        "rules": [
          {
            "state": "firing",
            "name": "alert_0",
            "query": "vector(1) > 0",
            "duration": 60,
            "keepFiringFor": 0,
            "labels": {
              "severity": "no"
            },
            "annotations": {
              "summary": "Always firing"
            },
            "alerts": null,
            "health": "ok",
            "lastError": "",
            "type": "alerting",
            "lastEvaluation": "2024-09-17T12:17:31.722263334Z",
            "evaluationTime": 0.005084959
          }
        ],
        "interval": 30,
        "lastEvaluation": "2024-09-17T12:17:31.722165334Z",
        "evaluationTime": 0.017278292,
        "sourceTenants": null
      }
    ]
  },
  "errorType": "",
  "error": ""
}
  ```
</details>

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/alerting-squad/issues/912
Part of https://github.com/grafana/alerting-squad/issues/871

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: docs/sources/mimir/references/http-api/index.md [hunk 1]
```diff
@@ -824,6 +824,8 @@ The `type` parameter is optional. If set, only the specified type of rule is ret
 
 The `file`, `rule_group` and `rule_name` parameters are optional, and can accept multiple values. If set, the response content is filtered accordingly.
 
+The `exclude_alerts` parameter is optional. If set, only rules are returned, active alerts are excluded.
```
Comment (tacole02) [inline docs/sources/mimir/references/http-api/index.md:827]:
```suggestion
The `exclude_alerts` parameter is optional. If set, it only returns rules and excludes active alerts.
```

**Review by tacole02 — APPROVED** [review]
LGTM 😄

File: CHANGELOG.md [hunk 2]
```diff
@@ -88,6 +88,7 @@
 * [ENHANCEMENT] Querier: attach logs emitted during query consistency check to trace span for query. #9213
 * [ENHANCEMENT] Query-scheduler: Experimental `-query-scheduler.prioritize-query-components` flag enables the querier-worker queue priority algorithm to take precedence over tenant rotation when dequeuing requests. #9220
 * [ENHANCEMENT] Add application credential arguments for Openstack Swift storage backend. #9181
+* [ENHANCEMENT] Support `exclude_alerts` parameter in `<prometheus-http-prefix>/api/v1/rules` endpoint. #9300
```
Comment (stevesg) [inline CHANGELOG.md:91]:
```suggestion
* [ENHANCEMENT] Ruler: Support `exclude_alerts` parameter in `<prometheus-http-prefix>/api/v1/rules` endpoint. #9300
```

**Review by stevesg — APPROVED** [review]
LGTM, thanks! Just a nit in the Changelog.

**Pull request merged**